nimbus/tests/test_updates.rs

```diff
@@ -0,0 +1,201 @@
+            app_id: "fenix".to_string(),
+            ..Default::default()
+        };
+        NimbusClient::new(ctx, tmp_dir.path(), Some(config), aru)
+    }
+
+    fn startup(client: &mut NimbusClient, first_run: bool) -> Result<()> {
+        if first_run {
+            client.set_experiments_locally(initial_experiments())?;
+        }
+        client.apply_pending_updates()?;
+        client.fetch_experiments()?;
+        Ok(())
+    }
+
+    #[test]
+    fn test_two_phase_update() -> Result<()> {
+        let mut client = new_client("test_two_phase_update")?;
+        client.fetch_experiments()?;
+
+        // We have fetched the experiments from the server, but not put them into use yet.
+        // The experiments are pending.
+        let experiments = client.get_all_experiments()?;
+        assert_eq!(experiments.len(), 0);
+
+        // Now, the app chooses when to apply the pending updates to the experiments.
+        let events: Vec<_> = client.apply_pending_updates()?;
+        assert_eq!(events.len(), 1);
+
+        let experiments = client.get_all_experiments()?;
+        assert_eq!(experiments.len(), 1);
+        assert_eq!(experiments[0].slug, "secure-gold");
+
+        // Next time we start the app, we immediately apply pending updates,
+        // but there may not be any waiting.
+        let events: Vec<_> = client.apply_pending_updates()?;
+        // No change events
+        assert_eq!(events.len(), 0);
+
+        // Confirm that nothing has changed.
+        let experiments = client.get_all_experiments()?;
+        assert_eq!(experiments.len(), 1);
+        assert_eq!(experiments[0].slug, "secure-gold");
+
+        Ok(())
+    }
+
+    fn assert_experiment_count(client: &NimbusClient, count: usize) -> Result<()> {
+        let experiments = client.get_all_experiments()?;
+        assert_eq!(experiments.len(), count);
+        Ok(())
+    }
+
+    #[cfg(feature = "rkv-safe-mode")]
+    #[test]
+    fn test_set_experiments_locally() -> Result<()> {
+        let client = new_client("test_set_experiments_locally")?;
+        assert_experiment_count(&client, 0)?;
+
+        client.set_experiments_locally(initial_experiments())?;
+        assert_experiment_count(&client, 0)?;
+
+        client.apply_pending_updates()?;
+        assert_experiment_count(&client, 2)?;
+
+        client.set_experiments_locally(no_experiments())?;
+        assert_experiment_count(&client, 2)?;
+
+        client.apply_pending_updates()?;
+        assert_experiment_count(&client, 0)?;
+
+        Ok(())
+    }
+
+    #[cfg(feature = "rkv-safe-mode")]
+    #[test]
+    fn test_startup_behavior() -> Result<()> {
+        let mut client = new_client("test_startup_behavior")?;
+        startup(&mut client, true)?;
+
+        let experiments = client.get_all_experiments()?;
+        assert_eq!(experiments.len(), 2);
+        assert_eq!(experiments[0].slug, "secure-gold");
+        assert_eq!(experiments[1].slug, "startup-gold");
```

Comment:
I am puzzled why this is not in the order given.

Reply:
TIL that rkv iterators "returns the (key, value) tuples in order", which seems to explain it.